[MRG] Docstring checks for svm and covariance modules

NicolasHug opened the pull request:
<!--
Thanks for contributing a pull request! Please ensure you have taken a look at
the contribution guidelines: https://github.com/scikit-learn/scikit-learn/blob/master/CONTRIBUTING.md#pull-request-checklist
-->

#### Reference Issues/PRs

Fixes parts of #9325

<!--
Example: Fixes #1234. See also #3456.
Please use keywords (e.g., Fixes) to create link to the issues or pull requests
you resolved, so that they will automatically be closed when your pull request
is merged. See https://github.com/blog/1506-closing-issues-via-pull-requests
-->


#### What does this implement/fix? Explain your changes.

Removed some modules from the ignored modules list, and fixed following errors (mostly argument order)

#### Any other comments?

<!--
Please be aware that we are a loose team of volunteers so patience is
necessary; assistance handling other issues is very welcome. We value
all user contributions, no matter how minor they are. If we are slow to
review, either the pull request needs some benchmarking, tinkering,
convincing, etc. or more likely the reviewers are simply busy. In either
case, we ask for your understanding during the review process.
For more information, see our FAQ on this topic:
http://scikit-learn.org/dev/faq.html#why-is-my-pull-request-not-getting-any-attention.

Thanks for contributing!
-->

NicolasHug committed:
Ordered docstring params for svm/classes.py

NicolasHug committed:
Removed random_projection from ignored modules

NicolasHug committed:
Removed cross_decomposition from ignored modules

NicolasHug committed:
Fixed doc issues for covariance module

amueller commented:
Looks like pep8 is failing because some of the type descriptions are too long.
Those are kinda tricky. Sometimes using ``\`` works but usually we opt for just making the type description shorter, I think.

amueller commented on sklearn/svm/classes.py:
```diff
@@ -260,26 +260,22 @@ class LinearSVR(LinearModel, RegressorMixin):
+        Penalty parameter C of the error term. The penalty is a squared
+        l2 penalty. The bigger this parameter, the less regularization is used.
+
+    loss : string, 'epsilon_insensitive' or 'squared_epsilon_insensitive' (default='epsilon_insensitive')
```
hm ok looks like that was to long before. Maybe just keep ``string``, default='epsilon_insensitive' in the type line and put the rest in the description.

NicolasHug committed:
Fixed pep8 issue.

Also modified docstring to make it clearer that parameters 'l1' and 'l2'
are deprecated

NicolasHug committed:
removed space after parenthesis

jnothman reviewed (approved):
Nice!
